cmd/kubelet/app/server.go

```diff
@@ -806,8 +807,21 @@ func buildKubeletClientConfig(s *options.KubeletServer, nodeName types.NodeName)
 	}
 
 	kubeClientConfigOverrides(s, clientConfig)
+	closeAllConns, err := updateDialer(clientConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	return clientConfig, closeAllConns, nil
+}
 
-	return clientConfig, nil, nil
+// updateDialer instruments a restconfig with a dial. the returned function allows forcefully closing all active connections.
+func updateDialer(clientConfig *restclient.Config) (func(), error) {
```

Comment:
this loses the safety checks UpdateTransport did to ensure an existing transport or dial option on the config was not overridden

Reply:
yeah, you can't override a dialer

Reply:
I think we need to move connrotation into rest client at a low level

Reply:
> this loses the safety checks UpdateTransport did to ensure an existing transport on the config was not set

Done, please take another look.



> I think we need to move connrotation into rest client at a low level

Yes, I agree with you. I will move this package in another PR